### DISCUSSION TITLE
Invalid hook call when configuring zustand

### DISCUSSION THREAD

**Discussion by im-ghost** [vercel/next.js#41236] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

![Screenshot_20221006-144240](https://user-images.githubusercontent.com/96427062/194328755-8f4caf51-0275-4466-9431-094a785800e6.png)

### Which example does this report relate to?

https://github.com/vercel/next.js/blob/canary/examples/with-zustand/

### What browser are you using? (if relevant)

Android Chrome

### How are you deploying your application? (if relevant)

next build

### Describe the Bug

When trying to configure with zustand,I get an issue saying I can't call useStore(Invalid hook call)

### Expected Behavior

To be able to use Zustand 

### To Reproduce

After the page renders,the issue pops up

**Comment by lucasmerlin** [comment]
I'm seeing the same problem with Next 13.0.0. It only happens when the appDir is enabled in next.config.js.
I also created a issue at the zustand repo: https://github.com/pmndrs/zustand/issues/1395 as well as a stackblitz example: https://stackblitz.com/edit/nextjs-egfcaj

**Reply by tlrjs** [reply]
This is happening for me even though appDir is not enabled.

**Reply by adriannecris** [reply]
Same for me, even without appDir
